### ISSUE TITLE
Bug: Shares are not being loaded

### ISSUE THREAD

**Issue by PhMemmel** [owncloud/moodle-repository_ocis#70]
Reproduction steps:

- Log into ocis as user1
- Share file to user2
- Log into moodle as user2, open repository file picker after connecting account of user2
- Under "Shares" you will not find the share

In fact I can only see a "New Folder" under "Shares" which I do not know where it comes from. Some old cached content?

Trying to logout from the file picker and re-login to maybe get rid off the caching problem suddenly lead to the following error:

![grafik](https://github.com/owncloud/moodle-repository_ocis/assets/65113153/9e178b2b-6528-48df-ba70-1cc14657cf12)

After that the filepicker worked anyway, but the content of "Shares" is still wrong.

**Comment by individual-it** [comment]
Yes, this looks like an issue on ocis itself

**Comment by PhMemmel** [comment]
Alright, thank you for the feedback. You probably are only referring to the exception, not the basic bug about the wrong Shares, right? :)

**Comment by individual-it** [comment]
Yes, the exception is an ocis issue.

For the issue that the share is not visible, as receiver of the share, did you "Enable Sync"?
![grafik](https://github.com/owncloud/moodle-repository_ocis/assets/2425577/89134a80-211b-4c3a-8fe7-472bcc420723)

**Comment by PhMemmel** [comment]
I did not enable it explicitly, but it is enabled, yes.

![grafik](https://github.com/owncloud/moodle-repository_ocis/assets/65113153/71518021-abfa-4485-8519-226d49317e11)

**Comment by individual-it** [comment]
The screen shows it being disable, please try to click on "Synchronization aktivieren"

**Comment by PhMemmel** [comment]
You are right. I misunderstood the "check" icon as "sync is active". Maybe this should be turned into another icon... but not a moodle plugin issue :)

You are right, if I enable sync, the shares are properly synced into the moodle repository. This however is very bad UX, I cannot explain my users that they will only see shares for which they enabled synchronisation manually.
Is there a solution to avoid this? Enable autosync or anything like that?

Also there still is the issue, that I have an (probably old) share in the listing in the file picker. So the synchronisation for deleted shares probably is not working?

**Comment by individual-it** [comment]
ocis 5.0 should set the sync-status automatically to 'synced' if `FRONTEND_AUTO_ACCEPT_SHARES` is not disabled

we also don't cache any responses, so the problem with the not accessible folder need to be debugged on ocis

**Comment by PhMemmel** [comment]
Well, the folders and files I shared were immediately available in the web UI, I did not have to accept the shares, so I guess FRONTEND_AUTO_ACCEPT_SHARES seems to be enabled. There was obviously no auto syncing however.

**Comment by individual-it** [comment]
## Auto sync/accept
sync & accept is the same thing. "accept" is just the old word for that. The idea is AFAIK that the user would receive a share with a lot of data, but by not syncing/accepting it refuse to get it synced to the desktop etc.
The moodle plugin currently follows the same logic. If the share is not synced/accepted its not shown in the moodle plugin.

## ghost-share folder
I believe the "New Folder" that you are missing is a "hidden" share.
In the WebUI you can "hide" shares from that view. This will only hide them from the web-view, nothing else.

Please try to click on "Hidden Shares" / "Ausgeblendete Freigaben" and check if you can find it there. You can "unhide" it again.
But that state is IMO not meant to have any influence to the moodle plugin.

![grafik](https://github.com/owncloud/moodle-repository_ocis/assets/2425577/17339d31-14a8-4b9c-8d07-d8ed9f6f7d82)
![grafik](https://github.com/owncloud/moodle-repository_ocis/assets/2425577/3e33fbf3-bba6-415f-848d-b2a9d9d5dba7)

**Comment by PhMemmel** [comment]
You are right. It was a hidden share, so the "Shares" feature works. Thank you for the explanation.

I agree, the moodle plugin should also show the hidden shares, the filter should not apply there. So everything's fine, the user just needs to know about the hidden share filter which he/she usually does, because he explicitely hid the share himself/herself I guess.

It remains the question if the moodle plugin should follow the sync logic at all (and if it's technically possible to ignore this): Even though the authentication mechanism behaves like a separate device connecting to ocis, for the user the moodle plugin just feels like a file picker. IMHO what is shown and what not should therefore be pretty equal to what the webview shows. Do you agree?
